src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt

```diff
@@ -274,16 +357,24 @@ class ClusterDetailFragment : Fragment() {
 
   companion object {
     private const val TAG = "ClusterDetailFragment"
-    private const val ENDPOINT_ID_KEY = "endpoint_id"
-    private const val DEVICE_PTR_KEY = "device_ptr"
+    private const val ENDPOINT_ID_KEY = "endpointId"
+    private const val HISTORY_COMMAND = "historyCommand"
+    private var clusterMap: Map<String, ClusterInfo> = ClusterInfoMapping().clusterMap
+    private lateinit var selectedClusterInfo: ClusterInfo
+    private lateinit var selectedCluster: ChipClusters.BaseChipCluster
+    private lateinit var selectedCommandCallback: DelegatedClusterCallback
+    private lateinit var selectedInteractionInfo: InteractionInfo
+    private lateinit var historyCommand: HistoryCommand
+    private var devicePtr = ClusterInteractionFragment.devicePtr
+    var endpointId by Delegates.notNull<Int>()
```

Comment:
Yes these should revert back to be just variables. However, there is no way to know the endpointId in historyTab, so I use endpointId as a static variable inside clusterDetailFragment.

Reply:
Using a mutable static variable in this manner is typically bad practice because it can be randomly modified by any class in the app. Maybe we could store the endpoint id within `HistoryCommand` instead? That would make more sense anyways since commands can be sent to different endpoints.